examples/reflection/reflection.rs

```diff
@@ -65,13 +65,20 @@ fn setup(type_registry: Res<AppTypeRegistry>) {
     assert_eq!(value.a, 2);
     assert_eq!(*value.get_field::<usize>("a").unwrap(), 2);
 
-    // You can also get the &dyn Reflect value of a field like this
+    // You can also get the `&dyn PartialReflect` value of a field like this
     let field = value.field("a").unwrap();
 
-    // you can downcast Reflect values like this:
-    assert_eq!(*field.downcast_ref::<usize>().unwrap(), 2);
+    // But values introspected via `PartialReflect` will not return `dyn Reflect` trait objects
+    // (even if the containing type does implement `Reflect`), so we need to convert them:
```

Comment:
I think we need a bit more information on `PartialReflect` vs `Reflect` here. Not all the details like you might find in the docs, but just a brief one or two sentence comment about what `PartialReflect` is and why we might want to convert to `Reflect`.